### PR TITLE
feat(vertical-nav): add tokens and update shim

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-788

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss [hunk 1]
```diff
@@ -449,8 +552,7 @@
 
       .nav-icon {
         //Dims
-        margin: 0;
-        margin-left: variables.$clr_baselineRem_0_667;
+        margin: var(--clr-vertical-nav-collapsed-item-margin, 0 0 0 variables.$clr_baselineRem_0_667);
```
Comment (valentin-mladenov) [inline projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss:555]: NO need for mixin?

**Review by valentin-mladenov — COMMENTED** [review]
Check my comments

File: projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss [hunk 2]
```diff
@@ -345,11 +427,17 @@
     .nav-content > .nav-link {
       //Display
       flex: 0 0 vertical-nav-variables.$clr-vertical-nav-item-height;
+      @if variables.$clr-use-custom-properties == true {
+        flex: 0 0 var(--clr-vertical-nav-item-height, vertical-nav-variables.$clr-vertical-nav-item-height);
+      }
```
[the remark this answers is not in the document]
Comment (dtsanevmw) [reply projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss:432]: It's basically the same and easier for cleaning after that, not that it's not possible.

File: projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss:555]: Missed that, thanks!

**Review by valentin-mladenov — APPROVED** [review]
LGTM

**Pull request merged**